### PR TITLE
Add documentation examples for `non-parent-init-called`

### PR DESCRIPTION
**Pull request by julthep**
Co-authored-by: Vladyslav Krylasov <vladyslav.krylasov@gmail.com>

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description
Add documentation examples for `non-parent-init-called`

Refs #5953

### CODE REVIEW

File: doc/data/messages/n/non-parent-init-called/bad.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+class Rectangle(Shape):
+    def __init__(self):
+        super().__init__()
+        self.rectangle = True
+
+class Square(Rectangle):
+    def __init__(self):
+        Shape.__init__(self)  # [non-parent-init-called]
+        self.square = True
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/n/non-parent-init-called/bad.py:13]:
```suggestion
class Animal:
    def __init__(self):
        self.is_multicellular = True


class Vertebrate(Animal):
    def __init__(self):
        super().__init__()
        self.has_vertebrae = True


class Cat(Vertebrate):
    def __init__(self):
        Animal.__init__(self)  # [non-parent-init-called]
        self.adorable = True
```

File: doc/data/messages/n/non-parent-init-called/good.py [hunk 2]
```diff
@@ -1 +1,13 @@
...
+
+class Rectangle(Shape):
+    def __init__(self):
+        super().__init__()
+        self.rectangle = True
+
+class Square(Rectangle):
+    def __init__(self):
+        super().__init__()
+        self.square = True
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/n/non-parent-init-called/good.py:13]:
```suggestion
class Animal:
    def __init__(self):
        self.is_multicellular = True


class Vertebrate(Animal):
    def __init__(self):
        super().__init__()
        self.has_vertebrae = True


class Cat(Vertebrate):
    def __init__(self):
        Vertebrate.__init__(self)  # [non-parent-init-called]
        self.is_adorable = True
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
A little nitpick because the base class can't be the more specialized one, Shape must have a lot of behavior that are specific to it and Square is "simple" with only one attribute.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Review by Pierre-Sassoulas — COMMENTED** [review]
Thank you for all the work you're doing right now to document pylint @julthep , impressive !

**Comment by julthep** [comment]
> Thank you for all the work you're doing right now to document pylint @julthep , impressive !

You're welcome. Actually, not as impressive as all of you folks are doing wonderful work without breaks for this awesome project.

**Pull request merged**